src/main/kotlin/com/exactpro/th2/codec/MessageDatumReader.kt

```diff
@@ -35,14 +35,16 @@ import java.nio.ByteBuffer
 import javax.xml.bind.DatatypeConverter
 import mu.KotlinLogging
 
-class MessageDatumReader(schema: Schema, private val enableIdPrefixEnumFields: Boolean = false) :
-    GenericDatumReader<Message.Builder>(schema, schema, getData()) {
+class MessageDatumReader(
+    schema: Schema,
+    private val enableIdPrefixEnumFields: Boolean? = false,
+) : GenericDatumReader<Message.Builder>(schema, schema, getData()) {
     @Throws(IOException::class)
     override fun readWithoutConversion(old: Any?, expected: Schema, decoder: ResolvingDecoder): Any? {
         return if (expected.type == Schema.Type.UNION) {
             val readIndex = decoder.readIndex()
             val schema = expected.types[readIndex]
-            UnionData(read(old, schema, decoder), if(enableIdPrefixEnumFields) "$UNION_ID_PREFIX$readIndex" else schema.name)
+            UnionData(read(old, schema, decoder), enableIdPrefixEnumFields?.let { if(it) "$UNION_ID_PREFIX$readIndex" else schema.name })
```

Comment:
maybe extract `enableIdPrefixEnumFields?.let { if(it) "$UNION_ID_PREFIX$readIndex" else schema.name }` to a separate method and reuse in all implementation instead of sharing variable